src/WordPress/Hooks.php

```diff
@@ -146,28 +146,36 @@ public function purgeCacheByRelevantURLs($postId)
             }
             $wpDomain = $wpDomainList[0];
 
-            // Do not purge for autosaves or updates to post revisions.
-            if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
-                return;
-            }
+            $postIds = (array) $postIds;
+            $urls = [];
+            foreach ($postIds as $postId) {
+                // Do not purge for autosaves or updates to post revisions.
+                if (wp_is_post_autosave($postId) || wp_is_post_revision($postId)) {
+                    continue;
+                }
 
-            $postType = get_post_type_object(get_post_type($postId));
-            if (!is_post_type_viewable($postType)) {
-                return;
-            }
+                $postType = get_post_type_object(get_post_type($postId));
+                if (!is_post_type_viewable($postType)) {
+                    continue;
+                }
 
-            $savedPost = get_post($postId);
-            if (!is_a($savedPost, 'WP_Post')) {
-                return;
-            }
+                $savedPost = get_post($postId);
+                if (!is_a($savedPost, 'WP_Post')) {
+                    continue;
+                }
 
-            $urls = $this->getPostRelatedLinks($postId);
-            $urls = apply_filters('cloudflare_purge_by_url', $urls, $postId);
+                $urls = array_values(array_unique(array_merge($urls, $this->getPostRelatedLinks($postId))));
+                $urls = apply_filters('cloudflare_purge_by_url', $urls, $postId);
+            }
+            $urls = apply_filters('cloudflare_purge_by_urls', $urls, $postIds);
```

Comment:
isn't this doubling the work above? we're already iterating the `postIds` but here we're applying the filter...again?

Reply:
This is a different hook that has aggregated urls and all post ids, more like a job hook vs the singular cloudflare_purge_by_url that is ran for each post id. Would allow for user to check against all post ids and urls

Reply:
from what i can see, these the same functionality. they both act on `$postIds` - one happens in a loop and the outside the loop. the one inside the loop gets the related links but the other just applies the filter to all of the remain URLs again. i suspect if you remove the second one here, your functionality won't change.

Reply:
I did leave the second aggregate hook in the latest commit. Efficiency wise, if you wanted to know the full result of what was about to be cleared and which post_ids were involved, without "cloudflare_purge_by_urls" you would need to monitor the singular cloudflare_purge_by_url for possibly many many iterations (post ids) and save that data to a temporary location (session, global, db, etc) in order to achieve what you could achieve with a single filter and single routine.

Reply:
I'm looking this over again.  Moving the unique filtering inside of getPostRelatedLinks does make sense.  However, if you dont apply unique again after you merge with the previous post id related, you end up passing a lot of duplicates to the hook cloudflare_purge_by_url.  I don't think any user would expect that

Reply:
i'm not sure i follow. can you try explaining another way? 

i still don't think we need both of these `apply_filters` call; we do it inside of the loop and additionally afterwards as well. i cannot see why we would need to duplicate this given every request will pass through the loop and already fetch the related links.

Reply:
A common case may be where you clear the local cache when clearing cf cache pages? I e. wp_cache_flush. When it was a single call, the existing would work well, with multiple ids, we would end up clearing the local cache N times. I'm down to remove it, but it does seem that a single hook could be useful.  I'll try to consider some other cases

Reply:
Thinking about it more, most of the cases i can think of would be better suited for an action after the cache is actually cleared via api requests